### PR TITLE
fix: select: adds fall back and optional chaining to avoid null reference errors at runtime

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:

- Adds options fallback to empty array where needed and some optional chaining
- Updates `placeholder` to show when `options` are null in` multiple` mode
- Adds unit tests covering null or empty array scenarios
- Updates `options` to ensure `id` alignment.

https://github.com/EightfoldAI/octuple/assets/99700808/4e1e18c7-81da-4707-be4e-10a158edfa2c

## GITHUB ISSUE (Open Source Contributors)
https://github.com/EightfoldAI/octuple/issues/666

## JIRA TASK (Eightfold Employees Only):
ENG-58585

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [ ] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [x] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Select` stories behave as expected when its `options` are `null` or `[]`.

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -110,11 +110,11 @@ export const Select: FC<SelectProps> = React.forwardRef(
     );
     const [dropdownVisible, setDropdownVisibility] = useState<boolean>(false);
     const [options, setOptions] = useState<SelectOption[]>(
-      _options.map((option: SelectOption, index: number) => ({
+      _options?.map((option: SelectOption, index: number) => ({
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:113]: do you want to fallback to an empty array so code can rely on this being an array?
```suggestion
      (_options || []).map((option: SelectOption, index: number) => ({
```

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -147,12 +147,12 @@ export const Select: FC<SelectProps> = React.forwardRef(
 
     const getSelectedOptionValues = (): SelectOption['value'][] => {
       return options
-        .filter((option: SelectOption) => option.selected)
-        .map((option: SelectOption) => option.value);
+        ?.filter((option: SelectOption) => option?.selected)
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:150]: by falling back to array above you probably don't need these optional chains from here down

File: src/components/Select/Select.tsx [hunk 3]
```diff
@@ -161,20 +161,20 @@ export const Select: FC<SelectProps> = React.forwardRef(
...
     useEffect(() => {
-      const selected: SelectOption[] = options.filter(
-        (opt: SelectOption) => opt.selected
+      const selected: SelectOption[] = options?.filter(
+        (opt: SelectOption) => opt?.selected
       );
       setOptions(
-        _options.map((option: SelectOption, index: number) => ({
-          selected: !!selected.find((opt) => opt.value === option.value),
+        _options?.map((option: SelectOption, index: number) => ({
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:173]:
```suggestion
        (_options || []).map((option: SelectOption, index: number) => ({
```

File: src/components/Select/Select.tsx [hunk 4]
```diff
@@ -161,20 +161,20 @@ export const Select: FC<SelectProps> = React.forwardRef(
...
+          selected: !!selected?.find((opt) => opt?.value === option?.value),
           hideOption: false,
-          id: option.text + index,
-          object: option.object,
+          id: option?.text + '-' + index,
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:176]: perhaps loop functions can check that the type of option is correct and filter it out if not... things might go pretty sidewise if we are trying to support objects of the wrong type

File: src/components/Select/Select.tsx [hunk 5]
```diff
@@ -490,24 +490,24 @@ export const Select: FC<SelectProps> = React.forwardRef(
     };
 
     const getPills = (): JSX.Element => {
-      const selected: SelectOption[] = options.filter(
-        (opt: SelectOption) => opt.selected
+      const selected: SelectOption[] = options?.filter(
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:493]: similarly, if you provide an empty fallback, you can safely assume selected is an array below
```suggestion
      const selected: SelectOption[] = (options || []).filter(
```

File: src/components/Select/Select.tsx (hunk 4 above)
Comment (dkilgore-eightfold) [reply src/components/Select/Select.tsx:176]: This sounds risky, shouldn't JSON validation be deferred to the host app?

Comment (dkilgore-eightfold) [reply src/components/Select/Select.tsx]: A few risks:

- Unnecessary parsing
- Duplicate validation impacting perf
- Runtime error if validation is missing some edge case

**Review by jhoward-eightfold — APPROVED** [review]

**Pull request merged**